modules/http/src/main/java/org/mule/module/http/internal/listener/MalformedUrlRequestHandler.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.http.internal.listener;
+
+import static org.mule.module.http.api.HttpConstants.HttpStatus.BAD_REQUEST;
+
+/**
+ * Request handle for request calls to a malformed path.
+ */
+public class MalformedUrlRequestHandler extends ErrorRequestHandler
```

Comment:
Rename to `BadRequestHandler` which is more generic. We could then reuse it for similar scenarios we might encounter in the future.